### PR TITLE
Add colors to address bar

### PR DESCRIPTION
**Pull request by DJDavid98**
This completes the following points of #6:

 - Auto truncation of `http:` protocol from address bar
 - URL highlighting (green for https, black for domain, dark gray for the rest)

Newly added JavaScript code is well-commented, address part colors are specified in `browser.css`

### CODE REVIEW

**Comment by jdalton** [comment]
Awesome! :sparkles:

File: JSBrowser/js/components/address-bar.js [hunk 1]
```diff
@@ -137,9 +137,41 @@
         style.display = isHidden ? "inline-block" : "none";
     };
 
+    this.colorizeAddress = address => {
+        let HTML = '';
+        // Parse URL using an <a> element
+        let a = document.createElement('a');
```
Comment (jdalton) [inline JSBrowser/js/components/address-bar.js]: For URL parsing you can use the [Windows.Foundation.Uri](https://www.google.com/search?q=Windows.Foundation.Uri&oq=Windows.Foundation.Uri&aqs=chrome..69i57j0l4.656j0j7&sourceid=chrome&es_sm=93&ie=UTF-8) WinRT API.

We have the `URI` const set [here](https://github.com/DJDavid98/JSBrowser/blob/4f2d59a04b09ac1cbcccb6cbffa7e4149df393c3/JSBrowser/js/components/address-bar.js#L7).
You can see a usage example [here](https://github.com/DJDavid98/JSBrowser/blob/4f2d59a04b09ac1cbcccb6cbffa7e4149df393c3/JSBrowser/js/components/address-bar.js#L7)

File: JSBrowser/css/browser.css [hunk 2]
```diff
@@ -24,32 +24,76 @@ html {
...
+    white-space: nowrap;
+    overflow: hidden;
+}
+#urlDisplay .https {
+    color: #090;
+}
+#urlDisplay .domain {
+    color: #000 !important;
+}
+#urlDisplay:after {
```
Comment (jayrenn) [inline JSBrowser/css/browser.css]: What is this used for?

File: JSBrowser/css/browser.css [hunk 3]
```diff
@@ -24,32 +24,76 @@ html {
...
+    pointer-events: none;
+    border-color:  transparent;
+    white-space: nowrap;
+    overflow: hidden;
+}
+#urlDisplay .https {
+    color: #090;
+}
+#urlDisplay .domain {
+    color: #000 !important;
```
Comment (jayrenn) [inline JSBrowser/css/browser.css]: Is there any way to not use the `!important` flag?

File: JSBrowser/js/components/address-bar.js [hunk 4]
```diff
@@ -137,9 +137,41 @@
...
+            ? a.hostname.replace(hostnameRegex, "$1<span class='domain'>$2</span>")
+            : "<span class='domain'>" + a.hostname + "</span>";
+
+        // Add port if it's a valid number
+        if (a.port.length > 0 && !isNaN(a.port))
+            HTML += ':' + a.port;
+
+        // Remove last / from pathname if original address doesn't have it either
+        let pathname = a.pathname;
+        if (!/\/$/.test(address))
```
Comment (jdalton) [inline JSBrowser/js/components/address-bar.js]: You could use `String#endsWith`.

**Comment by jdalton** [comment]
Instead of displaying the full `http:` protocol when you select the text box it should remain hidden and only show up if they copy the URL bar.

File: JSBrowser/css/browser.css (hunk 2 above)
Comment (DJDavid98) [inline JSBrowser/css/browser.css]: If the URL overflows the bar this gradient fades out the end of it

**Comment by DJDavid98** [comment]
Got rid of `!important`, utilized the built-in URI parser and made the `http://` part invisible in the input. I tried adding it back properly when copying but I'd need to hook into a "copy" event and I'm not sure of there's one.

**Comment by jayrenn** [comment]
Great stuff! Regarding the copy event, have you looked into using the [Clipboard.ContentChanged](https://msdn.microsoft.com/en-us/library/windows/apps/windows.applicationmodel.datatransfer.clipboard.contentchanged.aspx) WinRT API?

**Comment by DJDavid98** [comment]
@jayrenn Hooking into a central event tracking *_all_* clipboard changes would not be ideal in my opinion since it'd be hard to know when the copied content actually comes from the address bar or just some content being copied from the page. The best would be if there was a way to copy arbitrary text content, then I could hook into the `keyup` event of `#urlInput`, check when Ctrl+C is pressed, intercept it, and copy the URL (with `http://` added if necessary) to the clipboard. I tried doing this with `document.execCommand('copy')` but it didn't do anything. This would definitely not work with the Right Click > Copy option, but it'd be something at least.

**Comment by jayrenn** [comment]
The [`oncopy` event](http://www.w3schools.com/jsref/event_oncopy.asp) allows you to tell when a user is copying text from the address bar, specifically. Perhaps that event would trigger a clipboard management API?

**Comment by DJDavid98** [comment]
Intercepting an event is not an issue, putting any kind of custom text data on the clipboard itself is. Doesn't matter if I caught `keydown` or `copy`, I'd still need to alter the data that gets copied.

**Comment by jdalton** [comment]
Hi @DJDavid98! Sorry for the delay can you resubmit the PR. We have a CLA bot in place now and think it should kick in with a fresh request.

**Comment by jdalton** [comment]
Thank you!

**Comment by DJDavid98** [comment]
No problem, I understand that this could've caused legal issues before and that's probably the reason for the delay.

**Pull request closed without merging**